#162685656 Integrate Coveralls code coverage service with badge

tomuhenry opened the pull request:
**What does this PR do?**
This PR delivers a chore that integrates Coveralls code coverage service

**Description of Task to be completed?**
Connecting and configuring _Coveralls_ with _GitHub_ and _Travis_ to have code test coverage after successful Travis build.

**What are the relevant pivotal tracker stories?**
[#162685649](https://www.pivotaltracker.com/story/show/162685656)

[Link to coveralls](https://coveralls.io/github/andela/Ah-backend-guardians)

**Screenshot:**
<img width="1374" alt="screen shot 2019-01-08 at 7 11 06 pm" src="https://user-images.githubusercontent.com/40206038/50843199-4b7d8a80-1379-11e9-8323-57d6806d81ee.png">

BarnaTB reviewed (changes requested):
Your PR has too many commits so you could adopt of either squashing your commits or amending them to previous ones.
Also looking at your `README`, your badge shows `unknown` as opposed to what is shown on the coveralls website itself so you might want to put the badge that maps to your exact branch since it looks like it is displaying the `develop` branch.

fahadmak reviewed (commented):
Try to squash the commits into one commit

tomuhenry commented:
I've done a rebase and merged the commits into one @fahadmak 
I have also made the change to the **readme** and added the badge for the right branch @BarnaTB

archbishop12 commented on .coveralls.yml:
```diff
@@ -0,0 +1 @@
+repo_token: 6ggWHxHTG2JvepQ98oe4baMzh7EWPOtDM
```
@tomuhenry Could you add this as an environment variable. Thanks.

tomuhenry replied:
Ok... Working on it

tomuhenry committed:
chore(coveralls): Integrate Coveralls with README badge

 - Add repo_token to travis environment variable online
 - Add coveralls badge to Readme.md file
 - Add coveralls for after-success in .travis.yml file

[Maintains #162685656]

archbishop12 commented on .travis.yml:
```diff
@@ -4,6 +4,9 @@ python:
   postgresql: "9.4"
 
+env:
+    - repo_token: 6ggWHxHTG2JvepQ98oe4baMzh7EWPOtDM
```
@tomuhenry Add it to the environment variables on travis, it should not be visible here. Thanks

fahadmak commented:
LGTM

VivianDoreen commented:
LGTO Great work @tomuhenry

BarnaTB reviewed (approved):
Great work Tom! LGTM.

tomuhenry committed:
chore(coveralls): Integrate Coveralls with README badge

 - Change README badge to reflect develop branch

[Maintains #162685656]
